lib/sender.js

```diff
@@ -347,7 +350,7 @@ class Sender {
       }
     } else {
       this.sendFrame(
-        Sender.frame(buf, {
+        Sender.frame(data, {
```

Comment:
If `toBuffer()` is skipped here, then `toBuffer.readOnly` might not be the expected one. For example, it might be `true` when writing a string leading to an unnecessary copy of the data when masking the data.

```js
ws.send(Buffer.from('foo'));
ws.send('bar');
```

Reply:
Fixed

Reply:
Now it ignores the `readOnly` info, so it might mutate a user provided `Buffer`

Reply:
now?

Reply:
Still ignored if `permessage-deflate` is disabled.

Reply:
So we only ignore it if `buf` is `string` which is only true if `!perMessageDeflate && typeof data === 'string'`

Reply:
I don't see a problem atm.